### ISSUE TITLE
`text` seems to miss certain characters, consistently

### ISSUE THREAD

**Issue by jasonjmcghee** [enigo-rs/enigo#260]
**Describe the bug**
When using `text` and passing a variety of strings to it (rapidly calling in this use case)

Demonstration in notes here, but it's a consistent issue across applications / places to type text.

https://github.com/jasonjmcghee/plock/assets/1522149/19746ac1-1dba-40d0-befe-d5a3d4b9b72a

**To Reproduce**
Read the piped output of a `cat` command of a large file in a chunked fashion, and pipe the output to the `text` function 

**Expected behavior**
I would expect all characters to be written.

**Environment:**

- OS: macOS
- Rust 1.75
- Library Version 0.2.0-rc2

**Additional context**

`enigo` plays a key role in a new project: https://github.com/jasonjmcghee/plock and I'd love to figure out a way to fix this!

**Comment by pentamassiv** [comment]
TLDR; try cloning the enigo repo and changing [line 340](https://github.com/enigo-rs/enigo/blob/main/src/macos/macos_impl.rs#L340) to a higher value (e.g 20).

Thank you for letting me know. I have a suspicion what the problem is. The `text` associated function needs to split up the given text in chunks of 20 characters and sends individual events for each chunk. This is due to an undocumented limitation of the `CGEventKeyboardSetUnicodeString` function. Unfortunately there needs to be a delay between input events in order for them to get registered. The length of the delay was determined by trial and error. My guess is that the delay is not enough when piping large amounts of text. It would be great if you could let me know if my suggestion helps with the problem and which value is high enough for you to reliably enter all text.

Pluck looks like an interesting project :+1: 
I briefly read the code related to `enigo` and noticed some comments where you mention more problems with `enigo`. Feel free to let me know about everything that caused problems and I'll (eventually) try to fix them :-)

**Comment by jasonjmcghee** [comment]
Thank you for taking the time to respond and the kind words!

Interesting! rdev had a similar comment in their code about sleeps between event simulation. Is there anyway to find out if the triggered event was successful? Artificial delays always feel like flakey solutions 😅

Would you consider making that 2ms delay a parameter on the struct?

I'll fork it and modify it and/or make it a param.

Really appreciate your time and this crate!

**Comment by jasonjmcghee** [comment]
As far as the other issue goes, Unicode parsing comes to mind. It kept panicking- I had to use raw key codes.

**Comment by kgoodridge** [comment]
@pentamassiv 

I was looking at this on plock. I found that if I made sure it never passed a string to enigo.text("\nbad") that started with a newline, it worked.

Try this (I'm on MacOS, in case it matters)

```
use enigo::{
     Enigo, Keyboard, Settings,
 };
use std::thread;
use std::time::Duration;

fn main() {
    let mut enigo = Enigo::new(&Settings::default()).unwrap();

    thread::sleep(Duration::from_secs(5));
    let _ = enigo.text("Hello, Enigo!\n\n");
    let _ = enigo.text("\n Why you no listen?");
    let _ = enigo.text("I know why\n");
    let _ = enigo.text("you don't like leading newlines");
    thread::sleep(Duration::from_secs(5));
}
```

**Comment by jasonjmcghee** [comment]
@pentamassiv 

The above didn't fix it for me, but changed things, which led me down a path. And whether or not this truly fixed it or just seemed to for everything I tried... This workaround was at the least, effective.

https://github.com/jasonjmcghee/plock/pull/22/commits/c432ee6956c1feaf3181fd13f82419a00d867108

I create chunks of length 19 and add a space before every newline. Feels like alchemy, but genuinely seemed to do the trick.

**Comment by pentamassiv** [comment]
Thank you both for the investigation. So far I have no idea why it does not work with a newline at the start.

**Comment by jasonjmcghee** [comment]
@pentamassiv  hate to be that guy but i don’t think that’s the right fix. That will cause newline characters to trigger submission on inputs. I’ll be required to switch to a fork after this commit

**Comment by pentamassiv** [comment]
Okay, no problem. Thank you for letting me know. I'll try to think of something else

**Comment by pentamassiv** [comment]
The issue happens if the string starts with one of these chars: `\n`, `\t`, `\r`
:S

**Comment by jasonjmcghee** [comment]
(sorry for lack of response- very busy times. I will test at some point soon and let you know!)

**Comment by jasonjmcghee** [comment]
For what it's worth, this seems to be working properly! Nice job @pentamassiv 🎉

**Comment by pentamassiv** [comment]
Great to hear. Thank you for testing it :+1:
